src/lib/services/feature-toggle-service.ts

```diff
@@ -378,17 +380,92 @@ class FeatureToggleService {
             parameters: featureStrategy.parameters,
             segments: segments.map((segment) => segment.id) ?? [],
         };
+
+        if (this.flagResolver.isEnabled('strategyVariant')) {
+            result.sortOrder = featureStrategy.sortOrder;
+        }
+        return result;
     }
 
     async updateStrategiesSortOrder(
-        featureName: string,
+        context: IFeatureStrategyContext,
+        sortOrders: SetStrategySortOrderSchema,
+        createdBy: string,
+        user?: User,
+    ): Promise<Saved<any>> {
+        await this.stopWhenChangeRequestsEnabled(
+            context.projectId,
+            context.environment,
+            user,
+        );
+
+        return this.unprotectedUpdateStrategiesSortOrder(
+            context,
+            sortOrders,
+            createdBy,
+        );
+    }
+
+    async unprotectedUpdateStrategiesSortOrder(
+        context: IFeatureStrategyContext,
         sortOrders: SetStrategySortOrderSchema,
+        createdBy: string,
     ): Promise<Saved<any>> {
+        const { featureName, environment, projectId: project } = context;
+        const existingOrder = (
+            await this.getStrategiesForEnvironment(
+                project,
+                featureName,
+                environment,
+            )
+        )
+            .sort((a, b) =>
+                !!a.sortOrder && !!b.sortOrder
```

Comment:
Can we reuse the sort done here: https://github.com/Unleash/unleash/pull/4218, it's a bit easier to read imo and while we should have strategies without sorting order, if we do, I'd rather they were handled in the same way

Reply:
+1